### PR TITLE
Create pull_request_template.md

### PR DESCRIPTION
**Pull request by jwestw**
This isn't linked to an issue. 

This PR should create a markdown template that will automatically appear in the first comment in each pull request that is made - which should help both the developer submitting the PR and the reviewer.

Template is adapted from: https://best-practice-and-impact.github.io/qa-of-code-guidance/peer_review.html

### CODE REVIEW

File: .github/pull_request_template.md [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+
+*Insert any instructions to help the reviewer, e.g. "install new requirements from `requirements.txt`"*
+
+#### Code
+
+- [ ] **Requirements** My/our code matches the requirements of the ticket
+- [ ] **Functionality**: New functions meet requirements in issue ticket
+- [ ] **Compliant Code** Code is as [PEP 8]([url](https://peps.python.org/pep-0008/)) compliant as I can humanly make it
+- [ ] **Clean Code** 
+    - [ ] Code has been linted
```
Comment (nkshaw23) [inline .github/pull_request_template.md:13]: Potentially add a bit about best way to do this e.g. install pylint in vs_code

File: .github/pull_request_template.md [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+
+#### Documentation
+
+Any new code includes all the following forms of documentation:
+
+- [ ] **Function Documentation** Docstrings within the function(s') definition(s) have been created
+    - [ ] Includes `parameters` and `returns` for all major functions 
+    - [ ] Includes data types
+- [ ] **Updated Documentation**: Documentation has been updated _or a new ticket to do so has been created_
+
```
Comment (nkshaw23) [inline .github/pull_request_template.md:24]: You mentioned this earlier I think - section on adding any data to sync if its required to run

File: .github/pull_request_template.md [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+---
+
+## Peer Review
+
+- [ ] All requirements install from (updated) `requirements.txt`
+- [ ] Documentation has been created and is clear
+- [ ] Doctrings has been created and accurately describes the function's functionality
+- [ ] Unit tests pass, or if not present _a new ticket to create tests has been created_
+
```
Comment (Antonio-John) [inline .github/pull_request_template.md:37]: I think a bullet like this in peer review. Maybe obvious but might be good to have as a check box.
-  [ ] Code runs successfully

File: .github/pull_request_template.md [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+#### Testing
+- [ ] **Unit tests** Unit tests have been created and are passing _or a new ticket to create tests has been created_
+
+---
+
+## Peer Review
+
+- [ ] All requirements install from (updated) `requirements.txt`
+- [ ] Documentation has been created and is clear
+- [ ] Doctrings has been created and accurately describes the function's functionality
```
Comment (paigeh1) [inline .github/pull_request_template.md:35]: Might also be worth writing what format we use for docstrings. Think we have to use [Google style docstrings](https://sphinxcontrib-napoleon.readthedocs.io/en/latest/example_google.html) for them to render in github pages.

**Review by paigeh1 — CHANGES_REQUESTED** [review]
Looks good, left one comment and agree with both @Antonio-John and @nkshaw23 on their comments.

**Comment by jwestw** [comment]
Should also have a bullet point about documentation (at least notes on what's been changed)

File: .github/pull_request_template.md (hunk 1 above)
Comment (james-westwood) [reply .github/pull_request_template.md:13]: I think this should be in the "For developers" section of the readme

File: .github/pull_request_template.md (hunk 2 above)
Comment (james-westwood) [reply .github/pull_request_template.md:24]: Done!

File: .github/pull_request_template.md (hunk 3 above)
Comment (james-westwood) [reply .github/pull_request_template.md:37]: Done

File: .github/pull_request_template.md [hunk 5]
```diff
@@ -0,0 +1,71 @@
+## Pull Request submission 
+
+*Insert detailed bullet points about your changes here!*
+
+*Insert any instructions to help the reviewer, e.g. "install new requirements from `requirements.txt`"*
+
+*Let the reviewer know what data files are needed (to be grabbed from sync)
+
+#### Closes or fixes
```
Comment (Antonio-John) [inline .github/pull_request_template.md:9]: Very small comment. Could you put some more detail in here e.g *Insert issue/card number this closes/fixes* please?

**Review by Antonio-John — APPROVED** [review]

**Review by paigeh1 — APPROVED** [review]
Happy to approve

**Pull request merged**